ext/session/tests/session_set_save_handler_type_error.phpt

```diff
@@ -0,0 +1,61 @@
+--TEST--
+Test session_set_save_handler() function: interface wrong
+--SKIPIF--
+<?php include('skipif.inc'); ?>
+--INI--
+session.name=PHPSESSID
+session.save_handler=files
+--FILE--
+<?php
+
+$validCallback = function () { return true; };
+$deprecatedCallback = function () { return 0; };
+$exceptionCallback = function () { return []; };
+
+ob_start();
+
+try {
+    $ret = session_set_save_handler($exceptionCallback, $validCallback, $validCallback, $validCallback, $validCallback, $validCallback);
+    session_start();
+} catch (TypeError $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $ret = session_set_save_handler($deprecatedCallback, $validCallback, $validCallback, $validCallback, $validCallback, $validCallback);
+    session_start();
+} catch (TypeError $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $ret = session_set_save_handler($validCallback, $exceptionCallback, $validCallback, $validCallback, $validCallback, $validCallback);
+    session_start();
+} catch (TypeError $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $ret = session_set_save_handler($validCallback, $deprecatedCallback, $exceptionCallback, $validCallback, $validCallback, $validCallback);
+    session_start();
+} catch (TypeError $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+ob_end_flush();
+
+?>
+--EXPECTF--
+Warning: session_start(): Failed to initialize storage module: user (path: ) in %s on line %d
```

Comment:
Maybe It's a bit weird that both a warning and an exception is thrown. I could suppress the warning, but it might come handy to detect which callback fails.

Reply:
I do think we should suppress those warnings. Especially if an explicit exception is thrown from the handler, there really shouldn't be additional warnings.

Reply:
I've just tried to suppress all those unneeded warnings/exceptions in my new commit.